src/controller/tag-controller.js

```diff
@@ -0,0 +1,40 @@
+import TagService from '../service/tag-service.js';
+
+class TagController {
+  async getTags(req, res, next) {
+    try {
+      const { page = 1, limit = 10, order = 'desc', search } = req.query;
+      const tags = await TagService.getTags(
+        parseInt(page, 10),
+        parseInt(limit, 10),
+        order,
+        search,
+      );
+
+      if (tags.length === 0) {
+        return res.status(404).json({ error: '조건에 맞는 태그가 없습니다.' });
+      }
+
+      res.status(200).json({ data: tags, total: tags.length });
```

Comment:
```suggestion
      res.status(200).json({ data: tags, total: totalCount });
```
`tags.length`로 받으면 현재 페이지의 개수만 알려주기 때문에 다른 페이지에 검색 결과값이 존재할 때 문제가 발생할 수 있을 것 같습니다!